test/dynamo/test_higher_order_ops.py

```diff
@@ -0,0 +1,351 @@
+# Owner(s): ["module: dynamo"]
+import re
+import unittest
+
+import torch
+
+import torch._dynamo.test_case
+from torch._dynamo.utils import counters
+from torch._ops import wrap
+
+
+class MockBackend:
+    def __init__(self):
+        self.graphs = []
+
+    def __call__(self, gm: torch.fx.GraphModule, example_inputs):
+        self.graphs.append(gm)
+        return gm.forward
+
+
+global_var = torch.randn(3)
+global_num = 3.14
+
+
+class TestHigherOrderOps(torch._dynamo.test_case.TestCase):
+    def test_no_freevars(self):
+        mock = MockBackend()
+
+        def f(x):
+            return wrap(lambda x: torch.sin(x), x)
+
+        x = torch.randn(3)
+        expected = f(x)
+        result = torch.compile(f, backend=mock)(x)
+
+        self.assertEqual(result, expected)
+        self.assertEqual(len(mock.graphs), 1)
```

Comment:
Use compilecounter for this :P

```
counter = CompileCounter()
compiled = torch.compile(func, backend=counter)
self.assertEqual(counter.frame_count, 2)
```

Reply:
frame_count should suffice for both counting the graphs and graph breaks, remove counters too.

Reply:
You also get opcounts, which this test/ these tests would benefit from

Reply:
I also want to assert on what the graphs look like. I could modify CompileCounter to also keep track of the graphs?

Reply:
Below you mentioned CompileCounterWithBackend, which I think does what I want.